### PR TITLE
Implementing "suspected" mechanic

### PR DESCRIPTION
**Pull request by theelk801**
Should be good to go

### CODE REVIEW

File: Mage.Sets/src/mage/cards/c/ConvenientTarget.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    @Override
+    public ConvenientTargetEffect copy() {
+        return new ConvenientTargetEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Optional
+                .ofNullable(source.getSourcePermanentOrLKI(game))
+                .filter(Objects::nonNull)
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/c/ConvenientTarget.java:86]: why are you filtering `Objects::nonNull` inside an `Optional.ofNullable`?

File: Mage.Tests/src/test/java/org/mage/test/cards/continuous/SuspectTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    @Test
+    public void testNoSuspect() {
+        addCard(Zone.BATTLEFIELD, playerA, offender);
+
+        setStopAt(1, PhaseStep.END_TURN);
+        execute();
+
+        setStrictChooseMode(true);
```
Comment (xenohedron) [inline Mage.Tests/src/test/java/org/mage/test/cards/continuous/SuspectTest.java:36]: doesn't this need to be before the `execute()` ?

File: Mage.Tests/src/test/java/org/mage/test/cards/continuous/SuspectTest.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, alibi, offender);
+
+        activateAbility(1, PhaseStep.POSTCOMBAT_MAIN, playerA, "{2}{B}");
+
+        setStrictChooseMode(true);
+        setStopAt(1, PhaseStep.END_TURN);
+        execute();
+
+        assertSuspected(offender, false);
```
Comment (xenohedron) [inline Mage.Tests/src/test/java/org/mage/test/cards/continuous/SuspectTest.java:121]: also assert no counter

File: Mage.Tests/src/test/java/org/mage/test/cards/continuous/SuspectTest.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    @Test
+    public void testAlibiRemoveSuspect() {
+        addCard(Zone.BATTLEFIELD, playerA, "Bayou", 3 + 3);
+        addCard(Zone.BATTLEFIELD, playerA, offender);
+        addCard(Zone.HAND, playerA, alibi);
+
+        activateAbility(1, PhaseStep.PRECOMBAT_MAIN, playerA, "{2}{B}");
+
```
Comment (xenohedron) [inline Mage.Tests/src/test/java/org/mage/test/cards/continuous/SuspectTest.java:97]: can check menace ability present in between

**Review by xenohedron — APPROVED** [review]
minor comments on `Optional` usage and making tests a little more robust, otherwise looks good to me

File: Mage.Sets/src/mage/cards/c/ConvenientTarget.java (hunk 1 above)
Comment (theelk801) [reply Mage.Sets/src/mage/cards/c/ConvenientTarget.java:86]: `ofNullable` means it can be null, so we have to check that it isn't

Comment (xenohedron) [reply Mage.Sets/src/mage/cards/c/ConvenientTarget.java:86]: I don't think that's how Optional works.

My IDE is flagging this lambda as always returning true.

image

File: Mage.Tests/src/test/java/org/mage/test/cards/continuous/SuspectTest.java (hunk 4 above)
Comment (theelk801) [reply Mage.Tests/src/test/java/org/mage/test/cards/continuous/SuspectTest.java:97]: I don't think that's necessary, the other tests already handle that

File: Mage.Sets/src/mage/cards/c/ConvenientTarget.java (hunk 1 above)
Comment (theelk801) [reply Mage.Sets/src/mage/cards/c/ConvenientTarget.java:86]: that's weird, mine doesn't do that and it's definitely how it's supposed to work

**Pull request merged**